crates/language/src/buffer.rs

```diff
@@ -1243,6 +1243,12 @@ impl Buffer {
         self.text.snapshot()
     }
 
+    /// Retrieve the buffer's text preserving the original line endings for LSP communication.
+    /// This ensures that language servers receive text with the same line endings as the file.
+    pub fn text_for_lsp(&self) -> String {
+        fs::chunks(&self.as_rope(), self.line_ending()).collect()
```

Comment:
`fs::chunks` looks very-very odd: I would expect `Rope` to have a similar API already.
If there's none, let's make one, add a doc explaining why it's different from the `Display` (we're changing that anyway to mention `\n` peculiarity) and add a small unit test that shows that different line endings can be used correctly to "join" the text.